File: Structural/PrivateClass/java/SalaryCalculator.java
import java.math.BigDecimal;

public class SalaryCalculator {

    private SalaryData salaryData;

    public SalaryCalculator(BigDecimal baseSalary, BigDecimal bonuses, BigDecimal overtime) {
        this.salaryData = new SalaryData(baseSalary, bonuses, overtime);
    }

    public BigDecimal getIncome() {
        return salaryData.getBasePay().add(salaryData.getBonuses().add(salaryData.getOvertime()));
    }

    class SalaryData {

        private BigDecimal basePay;
        private BigDecimal bonuses;
        private BigDecimal overtime;

        public SalaryData(BigDecimal basePay, BigDecimal bonuses, BigDecimal overtime) {
            this.basePay = basePay;
            this.bonuses = bonuses;
            this.overtime = overtime;
        }

        public BigDecimal getBonuses() {
            return bonuses;
        }

        public BigDecimal getOvertime() {
            return overtime;
        }

        public BigDecimal getBasePay() {
            return basePay;
        }

    }

    public static void main(String [] args) {
        SalaryCalculator salaryCalculator = new SalaryCalculator(new BigDecimal(1000), new BigDecimal(100), new BigDecimal(40));
        System.out.println(salaryCalculator.getIncome());
    }
}

File: Structural/PrivateClass/kotlin/SalaryCalculator.kt
import java.math.BigDecimal

class SalaryCalculator(baseSalary: BigDecimal, bonuses: BigDecimal, overtime: BigDecimal) {

    private val salaryData: SalaryData

    val income: BigDecimal
        get() = salaryData.basePay.add(salaryData.bonuses.add(salaryData.overtime))

    init {
        this.salaryData = SalaryData(baseSalary, bonuses, overtime)
    }

    internal inner class SalaryData(val basePay: BigDecimal, val bonuses: BigDecimal, val overtime: BigDecimal)
}

fun main(args: Array<String>) {
    val salaryCalculator = SalaryCalculator(BigDecimal(1000), BigDecimal(100), BigDecimal(40))
    println(salaryCalculator.income)
}